tests/contest/contest/src/tests/linux_masked_paths/masked_paths.rs

```diff
@@ -0,0 +1,225 @@
+    if let TestResult::Passed = res {
+        TestResult::Failed(anyhow!(
+            "expected error in container creation with invalid symlink, found no error"
+        ))
+    } else {
+        TestResult::Passed
+    }
+}
+
+fn check_masked_symlinks() -> TestResult {
+    // Deliberately create a masked symlink that points an invalid file,
+    // and expect an error.
+    let root = PathBuf::from("/");
+    let masked_symlink = "masked_symlink";
+    let masked_paths = vec![root.join(masked_symlink).to_string_lossy().to_string()];
+    let spec = get_spec(masked_paths);
+
+    let res = test_inside_container(spec, &CreateOptions::default(), &|bundle_path| {
+        use std::{fs, io};
+        let test_file = bundle_path.join(masked_symlink);
+        // ln -s ../masked-symlink ; readlink -f /masked-symlink; ls -L /masked-symlink
+        match std::os::unix::fs::symlink("../masked_symlink", &test_file) {
+            io::Result::Ok(_) => { /* This is expected */ }
+            io::Result::Err(e) => {
+                bail!("error in creating symlink, to {:?} {:?}", test_file, e);
+            }
+        }
+
+        let r_path = match fs::read_link(&test_file) {
+            io::Result::Ok(p) => p,
+            io::Result::Err(e) => {
+                bail!("error in reading symlink at {:?} : {:?}", test_file, e);
+            }
+        };
+
+        // It ensures that the symlink points not to exist.
+        match fs::metadata(r_path) {
+            io::Result::Ok(md) => {
+                bail!(
+                    "reading path {:?} should have given error, found {:?} instead",
+                    test_file,
+                    md
+                )
+            }
+            io::Result::Err(e) => {
+                let err = e.kind();
+                if let io::ErrorKind::NotFound = err {
+                    Ok(())
+                } else {
+                    bail!("expected not found error, got {:?}", err);
+                }
+            }
+        }
+    });
+
+    // If the container creation succeeds, we expect an error since the masked paths does not support symlinks.
+    if let TestResult::Passed = res {
+        TestResult::Failed(anyhow!(
+            "expected error in container creation with invalid symlink, found no error"
+        ))
+    } else {
+        TestResult::Passed
+    }
+}
+
+fn test_mode(mode: u32) -> TestResult {
+    let root = PathBuf::from("/");
+    let masked_device = "masked_device";
+    let masked_paths = vec![root.join(masked_device).to_string_lossy().to_string()];
+    let spec = get_spec(masked_paths);
+
+    test_inside_container(spec, &CreateOptions::default(), &|bundle_path| {
+        use std::os::unix::fs::OpenOptionsExt;
+        use std::{fs, io};
+        let test_file = bundle_path.join(masked_device);
+
+        let mut opts = fs::OpenOptions::new();
+        opts.mode(mode);
+        opts.create(true);
```

Comment:
I think `opts` isn't used, right?

Reply:
You are correct. It seems that `opts` is not being used in the code. Thank you for pointing that out!